src/txmempool.h

```diff
@@ -535,6 +536,7 @@ class CTxMemPool
 
     std::map<CService, uint256> mapProTxAddresses;
     std::map<CKeyID, uint256> mapProTxPubKeyIDs;
+    std::map<uint256, uint256> mapProTxBlsPubKeys;
```

Comment:
Not sure I understand. Why does the name not match?

The reason I chose to use the hash as key is historical. I implemented this before I kept track of hashes inside CBLSPublicKey (and friends). Before this, using BLS primitives as keys would be horribly slow. I'm also not sure if I really should add an `operator<` to the primitives which doesn't really compare the primitives but the hashes of these. I'll however at least add a commit that uses `.GetHash()` instead of doing fresh serialization.

Reply:
Ok, I see. But, I mean, still, it says mapProTx_BlsPubKeys_ and yet bls pubkeys are nowhere to be found in this map, only their hashes. And then, isn't `CBLSId` basically it, the hash of the bls pubkey? So this could be:
```
std::map<CBLSId, uint256> mapProTxBlsPubKeyIDs;
```
just like `mapProTxPubKeyIDs ` for ecdsa pubkeys. And yes, would have to implement `operator<` in this case... BUT this led me to another thought - if all `CBLSId` does is storing hash then why implement it via `CBLSWrapper` at all? See https://github.com/UdjinM6/dash/commit/1446e932909780da4e5be0657294fe99e297da66. And then it's trivial to apply changes I described above https://github.com/UdjinM6/dash/commit/05af5bc2db815c8f34e06d546ea8469e11f9c115.

Reply:
Ah now I see what you mean.

Regarding CBLSId: This came from the initial herumi based implementation, which had `blsId` as it's own type. This is NOT the ID of any key, it's actually just an arbitrary 256 bit number. This ID is used in the polynomial functions only and identifies a share. It's the X value in the polynomial evaluation and interpolation.

Reply:
Ah and what I forgot to add. If Chia BLS ever officially starts to support Shamir's Secret Sharing, they might end up implementing their own class for IDs. That's why I left CBLSId as it is now and also let it be a derived class from CBLSWrapper

Reply:
Hmmm... so that's just some random ID? I guess this was smth I was missing. Anyway... Let's just change the name e.g. `std::map<uint256, uint256> mapProTxBlsPubKeyHashes;` and keep other stuff untouched then?

Reply:
Yepp, done.